numbers/src/test/java/org/dicio/numbers/lang/fr/FrenchFormatter.java

```diff
@@ -0,0 +1,236 @@
+                        groupName += " " + NUMBER_NAMES_SHORT_SCALE.get(magnitude);
+                    }
+                }
+
+                groupNames.add(groupName);
+                ordi = false;
+            }
+
+            appendSplitGroups(result, groupNames);
+
+        } else {
+            boolean ordi = ordinal && numberIsWhole; // not ordinal if not whole
+            final List<Long> groups = Utils.splitByModulus(numberLong, 1000000);
+            final List<String> groupNames = new ArrayList<>();
+            for (int i = 0; i < groups.size(); ++i) {
+                final long z = groups.get(i);
+                if (z == 0) {
+                    continue; // skip 000000 groups
+                }
+
+                String groupName;
+                if (z < 1000) {
+                    groupName = subThousand(z, i == 0 && ordi);
+                } else {
+                    groupName = subThousand(z / 1000, false) + " thousand";
+                    if (z % 1000 != 0) {
+                        groupName += (i == 0 ? ", " : " ") + subThousand(z % 1000, i == 0 && ordi);
+                    } else if (i == 0 && ordi) {
+                        if (z / 1000 == 1) {
+                            groupName = "thousandth"; // remove "one" from "one thousandth"
+                        } else {
+                            groupName += "th";
+                        }
+                    }
+                }
+
+                if (i != 0) {
+                    final long magnitude = Utils.longPow(1000000, i);
+                    if (ordi) {
+                        // ordi can be true only for the first group (i.e. at the end of the number)
+                        if (z == 1) {
+                            // remove "one" from first group (e.g. "one billion, millionth")
+                            groupName = ORDINAL_NAMES_LONG_SCALE.get(magnitude);
+                        } else {
+                            groupName += " " + ORDINAL_NAMES_LONG_SCALE.get(magnitude);
+                        }
+                    } else {
+                        groupName += " " + NUMBER_NAMES_LONG_SCALE.get(magnitude);
+                    }
+                }
+
+                groupNames.add(groupName);
+                ordi = false;
+            }
+
+            appendSplitGroups(result, groupNames);
+        }
+
+        if (realPlaces > 0) {
+            if (number < 1.0 && (result.length() == 0 || "minus ".contentEquals(result))) {
+                result.append("zero"); // nothing was written before
+            }
+            result.append(" point");
+
+            final String fractionalPart = String.format("%." + realPlaces + "f", number % 1);
+            for (int i = 2; i < fractionalPart.length(); ++i) {
+                result.append(" ");
+                result.append(NUMBER_NAMES.get((long) (fractionalPart.charAt(i) - '0')));
+            }
+        }
+
+        return result.toString();        return null;
```

Comment:
```suggestion
        return result.toString();
```

Reply:
Well that's what I get for using vsc instead of IDEA x). Thanks